### PR TITLE
Fix negative free storage on android

### PR DESCRIPTION
**Pull request by aalzanki**
Test Plan:
 - there's no way for me to test this right now. I copied this from https://github.com/Connected-Journeys/react-native-device-info/commit/fd1b3ce9ce0fc2a7d9bd40a53a908162bf7c3a38
 - I'm going to create a build for robert to test this with
 - https://khanacademy.atlassian.net/browse/MOB-1254

### CODE REVIEW

**Comment by aalzanki** [comment]
It's a int overflow issue.

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 1]
```diff
@@ -201,11 +202,15 @@ public Integer getTotalDiskCapacity() {
...
 
   @ReactMethod
-  public Integer getFreeDiskStorage() {
+  public Long getFreeDiskStorage() {
     try {
-      StatFs external = new StatFs(Environment.getExternalStorageDirectory().getAbsolutePath());
-      return external.getAvailableBlocks() * external.getBlockSize();
-    } catch (Exception e) {
+      File path = Environment.getExternalStorageDirectory();
+      SatFs statFs = new StatFs(path.getPath());
```
Comment (hblumberg) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: Why did we switch from `getAbsolutePath` to `getPath` here?

**Review by hblumberg — COMMENTED** [review]
Wow, excellent catch! How'd you figure it out?

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java (hunk 1 above)
Comment (aalzanki) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: The PR I copied this from switched. Just did quick search, seems like people use `getAbsolutePath`.

**Comment by aalzanki** [comment]
@hblumberg I just searched for "negative" in the repo's issues haha

**Review by jaredly — APPROVED** [review]
looks reasonable.

**Pull request merged**